### PR TITLE
spicetify-cli: rename bin

### PR DESCRIPTION
**Pull request by uncenter**
## Description of changes

Changes the `meta.mainProgram` and output bin from `spicetify-cli` to `spicetify`. The CLI is intended to be used under the `spicetify` command: see https://spicetify.app/docs/advanced-usage/command-line-interface/, https://spicetify.app/docs/getting-started#updating. I believe Homebrew also renames the binary to `spicetify`.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by SuperSandro2000 — DISMISSED** [review]

File: pkgs/by-name/sp/spicetify-cli/package.nix [hunk 1]
```diff
@@ -28,19 +35,22 @@ buildGoModule rec {
...
   doInstallCheck = true;
   installCheckPhase = ''
-    $out/bin/spicetify-cli --help > /dev/null
+    $out/bin/${meta.mainProgram} --help > /dev/null
   '';
 
   passthru.tests.version = testers.testVersion {
     package = spicetify-cli;
-    command = "spicetify-cli -v";
+    command = "${meta.mainProgram} -v";
```
[the remark this answers is not in the document]
Comment (uncenter) [reply pkgs/by-name/sp/spicetify-cli/package.nix]: Actually it looks like we don't need that whole `command = ...` line anyway: https://github.com/NixOS/nixpkgs/pull/309531#discussion_r1591158082.

**Comment by Gerg-L** [comment]
Ideally we would fix this upstream

I wish we could set `-o spicetify` to the `ldflags` but that does not work at all for some reason...

if `spicetify` is only a symlink don't set `meta.mainProgram` to it

just `mv spicetify-cli spicetify` would be ok with me

**Comment by uncenter** [comment]
> just `mv spicetify-cli spicetify` would be ok with me

https://github.com/NixOS/nixpkgs/pull/309570#discussion_r1591119852

**Comment by Gerg-L** [comment]
> Hmm... well it is the intended command to be used?

Yes but `meta.mainProgram` is used programmatically by `nix run` and `lib.getExe`

no user should run `meta.mainProgram` manually

**Comment by uncenter** [comment]
> > Hmm... well it is the intended command to be used?
> 
> Yes but `meta.mainProgram` is used programmatically by `nix run` and `lib.getExe`
> 
> no user should run `meta.mainProgram` manually
> 
> It's just for the off chance that something doesn't resolve symlinks properly

Should I `mv spicetify-cli spicetify` and then symlink the other way around?

**Comment by Gerg-L** [comment]
> Should I `mv spicetify-cli spicetify` and then symlink the other way around?

Yeah that makes the most sense to me

> I wish we could set -o spicetify to the ldflags but that does not work at all for some reason...

turn out it's a upstream thing, buildGoModule uses `go install` which is *less flexible* than go build.

**Review by Gerg-L — APPROVED** [review]
Looks good

**Review by mdarocha — APPROVED** [review]

**Review by amozeo — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**